crates/ruff_python_formatter/resources/test/fixtures/import_black_tests.py

```diff
@@ -1,31 +1,34 @@
+#!/usr/bin/python3
+
 from __future__ import annotations
 
 import argparse
-import glob
-import os
-import pathlib
+from pathlib import Path
 
 
-def import_fixture(black_input: pathlib.Path, fixture_set: str, fixture_name: str):
-    output_directory = pathlib.Path(os.path.dirname(__file__)).joinpath("black").joinpath(fixture_set)
+def import_fixture(fixture: Path, fixture_set: str):
+    """
+    Imports a single fixture by writing the input and expected output to the black directory.
+    """
 
+    output_directory = Path(__file__).parent.joinpath("black").joinpath(fixture_set)
     output_directory.mkdir(parents=True, exist_ok=True)
 
-    fixture_path = output_directory.joinpath(fixture_name)
+    fixture_path = output_directory.joinpath(fixture.name)
     expect_path = fixture_path.with_suffix(".py.expect")
 
     with (
-        black_input.open("r", encoding="UTF8") as black_file,
-        fixture_path.open("w", encoding="UTF8") as fixture_file,
-        expect_path.open("w", encoding="UTF8") as expect_file
+        fixture.open("r") as black_file,
+        fixture_path.open("w") as fixture_file,
+        expect_path.open("w") as expect_file
     ):
-        lines = black_file.readlines()
+        lines = iter(black_file)
         expected = []
         input = []
 
-        for (i, line) in enumerate(lines):
+        for line in lines:
             if line.rstrip() == "# output":
-                expected = lines[i + 1:]
+                expected = list(lines)
```

Comment:
sorry, it did not realize you're reusing lines here

Reply:
Why sorry? It works like a charm. I don't have to use indices anymore, because the iterator now tracks all state :)